File: codebattle/js/fibonacci.js
//Задача: fibonacci

//Return the nth item in the Fibonacci sequence. Hint: The first item in the sequence is 0.

//Example: 13 == solution(7)

//https://github.com/h/b/blob/master/src/b/issues/fibonacci.clj

module.exports = num=>{
  var fib = [0, 1];
  while (fib.length < num+1) {
    fib[fib.length] = fib[fib.length-1] + fib[fib.length-2]
  }
  
  console.log(num, fib)
  return fib[num]
}

File: codebattle/ruby/fibonacci.rb
#Задача: fibonacci

#Return the nth item in the Fibonacci sequence. Hint: The first item in the sequence is 0.

#Example: 13 == solution(7)

#https://github.com/Hexlet/battle_asserts/blob/master/src/battle_asserts/issues/fibonacci.clj

# BEGIN implement function `solution` here 
def solution(n)
  fib(0,1,n,0)
end

def fib (p,c,n,m)
  return c if m == n-1
  fib(c,c+p,n,m+1)
end
# END